src/app/js/fields/FieldComposedOf.js

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import SelectField from 'material-ui/SelectField';
+import Toggle from 'material-ui/Toggle';
+import MenuItem from 'material-ui/MenuItem';
+import FlatButton from 'material-ui/FlatButton';
+import IconButton from 'material-ui/IconButton';
+import IconDelete from 'material-ui/svg-icons/action/delete';
+import translate from 'redux-polyglot/translate';
+import compose from 'recompose/compose';
+import withHandlers from 'recompose/withHandlers';
+import { connect } from 'react-redux';
+import { change, formValueSelector } from 'redux-form';
+
+import { FIELD_FORM_NAME } from './';
+
+import {
+    polyglot as polyglotPropTypes,
+    field as fieldPropTypes,
+} from '../propTypes';
+
+const styles = {
+    inset: {
+        paddingLeft: 40,
+    },
+    compositionContainer: {
+        display: 'flex',
+    },
+    select: {
+        flexGrow: 2,
+    },
+    header: {
+        paddingLeft: 0,
+    },
+};
+
+const FieldComposedOf = ({
+    isComposedOf,
+    columns,
+    fields,
+    handleToggle,
+    handleSelectColumn,
+    handleAddColumn,
+    handleRemoveColumn,
+    p: polyglot,
+}) => (
+    <div>
+        <div style={styles.header}>{polyglot.t('wizard_composed_of')}</div>
+        <div style={styles.inset}>
+            <Toggle
+                label="is_composite_field"
+                toggled={isComposedOf}
+                onToggle={handleToggle}
+            />
+            {isComposedOf &&
+                columns.map((col, index) => (
+                    <div
+                        key={`composition_${index}`}
+                        style={styles.compositionContainer}
+                    >
+                        <SelectField
+                            className={`composite-field composite-field-${
+                                index
+                            }`}
+                            onChange={handleSelectColumn(index)}
+                            style={styles.select}
+                            hintText={polyglot.t('select_a_column')}
+                            value={col}
+                            required
+                        >
+                            {fields.map(f => (
+                                <MenuItem
```

Comment:
The JSX tends to be too indented and too long, split it into smaller components.